src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,35 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    if (animal instanceof Herbivore && !animal.hidden && animal.health > 0) {
+      animal.hide();
+      animal.health -= 50;
+
+      if (animal.health <=0) this.die();
```

Comment:
The `hide` method is being called inside the `bite` method, which is incorrect according to the checklist. The `hide` method should only set the `hidden` property to `true`, but it should not be called within the `bite` method. The `bite` method should only decrease the health of the herbivore if it's not hidden. Also, the `die` method is being called, but it is not defined anywhere in the `Carnivore` class. Instead of calling a `die` method, you should filter the `Animal.alive` array to remove the dead animal.